server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -72,6 +75,66 @@ public class DecommissionController {
         this.threadPool = threadPool;
     }
 
+    /**
+     * This method sends a transport call to retry decommission action, given that -
+     * 1. The request is not timed out
+     * 2. And executed when there was a cluster manager change
+     *
+     * @param decommissionRequest decommission request object
+     * @param startTime start time of previous request
+     * @param listener callback for the retry action
+     */
+    public void retryDecommissionAction(
```

Comment:
should we rename to `tryDecommissionOnNewClusterManager` as it is used only there for now ?

Reply:
The controller, is more of generic methods which can be used in the service layer. This method is just attempting a retry and hence the name. The place where we are using it is during leader switch (one use case of it.

Let me know if you think it makes sense. I can rename it if required